Use globstar to run notebooks in subdirs

pmrv opened the pull request:
Previously workflow doesn't run notebooks in sub directories, like

root/notebooks/foo/bar.ipynb

Using the `**` notation fixes this.  `EXCLUSION_FILE` should still work, but I haven't checked.

liamhuber reviewed (commented):
> Using the ** notation fixes this.  EXCLUSION_FILE should still work, but I haven't checked.

This seems like a good check to do before merging this -- it doesn't have to work, but at least I would hope something like `path_to/excluded_notebook.ipynb` does, in which case this behaviour + documentation update is totally sufficient.

Otherwise lgtm, and is a good change.

pmrv commented:
I've switched over to `find`, instead of weird shell extension I happen to know.  As expected there's no interaction with the exclusion file if it is given in relative paths and I've updated the docs accordingly.
